### DISCUSSION TITLE
VectorInput is insanely slow. Am I doing something wrong...?

### DISCUSSION THREAD

**Discussion by jamiebullock** [MTG/essentia#1363] [General]
I'm using Essentia in streaming mode, using the C++ API. 

If I feed my network with an `EasyLoader`, loading a 45 second WAV file, it takes less than a second to compute all the features. 

However, if I load the the WAV into memory (`std::vector<Real>`) and then use a `VectorInput<Real>` to feed my network, it takes over 60 seconds to compute the features!

The networks are of the form:

```
VectorInput<Real> *input = new VectorInput<Real>(&audio);
Algorithm *frame  = factory.create("FrameCutter", "frameSize", 2048, "hopSize", 1024);

input->output("data") >> frame->("signal");
frame->output("frame") >> // other stuff
```

It just gets slower as I add more nodes the network, whereas `EasyLoader` is almost instant.

Any ideas?

**Comment by nolan-veed** [comment]
Hey @jamiebullock 

My feeling is that your `VectorInput<Real>` is streaming it's output without much batching of audio samples.

The [`acquireSize`](https://github.com/MTG/essentia/blob/36ec3d929402d491cbbb1120c54631e8c205b0ff/src/essentia/streaming/algorithms/vectorinput.h#L29) defaults to 1, which means that your [`process()`](https://github.com/MTG/essentia/blob/36ec3d929402d491cbbb1120c54631e8c205b0ff/src/essentia/streaming/algorithms/vectorinput.h#L145) loop will be running more frequently - once per sample.

In the EasyLoader's [`Source<StereoSample>` instance](https://github.com/MTG/essentia/blob/36ec3d929402d491cbbb1120c54631e8c205b0ff/src/algorithms/io/audioloader.h#L95) is configured for larger sized buffers.

Let me know, if that helps!
